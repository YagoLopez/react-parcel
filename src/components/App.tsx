import * as React from 'react';
const puzzle = require('../puzzle.svg');

export default class App extends React.Component<{}, {}> {

  public render(): JSX.Element {
    return (
      <main className="container">
        <img src={ puzzle } className="logo" />
        <h1 className="title">React + TypeScript + Parcel</h1>
        <div>Starter Project</div>
      </main>
    );
  }
}
