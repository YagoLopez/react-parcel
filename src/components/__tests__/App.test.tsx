//todo: snapshots

import * as React from 'react';
import {mount, ReactWrapper} from 'enzyme';
import {App} from '../App';


describe("App component", () => {

  const component: any = mount(<App />);

  it("is defined", () => {
    expect(component).toBeDefined();
    expect(component).toBeInstanceOf(ReactWrapper);
  });

  it("renders main content", () => {
    // expect(component.find('main').length).toBe(1);
    expect(component.find('main').children().length).toBe(3);

  });

  // it("shows click count when toggled less than 4 times", () => {
  //   const wrap = mount(<App />);
  //   const toggleButton = wrap.find(".toggle button");
  //   toggleButton.simulate("click");
  //   toggleButton.simulate("click");
  // });

  // it("shows warning message when toggled more than 4 times", () => {
  //   const wrap = mount(<App />);
  //   const toggleButton = wrap.find(".toggle button");
  //   for (let i = 0; i <= 4; i += 1) {
  //     toggleButton.simulate("click");
  //   }
  //   expect(wrap.find(sel("click-warning")).exists()).toBeTruthy();
  // });

});
