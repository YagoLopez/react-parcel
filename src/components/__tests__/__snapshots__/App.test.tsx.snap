// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP COMPONENT matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="logo rotation"
          src="puzzle.svg"
        />,
        <h1
          className="title"
        >
          React + TypeScript + Parcel
        </h1>,
        <div>
          Starter Project
        </div>,
      ],
      "className": "container bordered",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "logo rotation",
          "src": "puzzle.svg",
        },
        "ref": "logo",
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "React + TypeScript + Parcel",
          "className": "title",
        },
        "ref": null,
        "rendered": "React + TypeScript + Parcel",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Starter Project",
        },
        "ref": null,
        "rendered": "Starter Project",
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="logo rotation"
            src="puzzle.svg"
          />,
          <h1
            className="title"
          >
            React + TypeScript + Parcel
          </h1>,
          <div>
            Starter Project
          </div>,
        ],
        "className": "container bordered",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo rotation",
            "src": "puzzle.svg",
          },
          "ref": "logo",
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React + TypeScript + Parcel",
            "className": "title",
          },
          "ref": null,
          "rendered": "React + TypeScript + Parcel",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Starter Project",
          },
          "ref": null,
          "rendered": "Starter Project",
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
